### PR TITLE
[RPaaS] Initial CLI Extensions

### PR DESCRIPTION
**Pull request by wwendyc**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
providerhub

**Comment by yungezz** [comment]
hi @wwendyc is this 3rd party LiftR RP?

**Comment by wwendyc** [comment]
hi @yungezz , we're not a part of Liftr. RPaaS is a 1st party RP: https://armwiki.azurewebsites.net/rpaas/overview.html

File: src/providerhub/azext_providerhub/generated/_params.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                                                            'TenantOnly', 'AuthorizationFree']), help='')
+        c.argument('required_features', nargs='+', help='')
+        c.argument('capabilities', action=AddCapabilities, nargs='+', help='')
+        c.argument('metadata', type=validate_file_or_dict, help='Any object Expected value: json-string/@json-file.')
+        c.argument('template_deployment_options', action=AddTemplateDeploymentOptions, nargs='+', help='')
+        c.argument('schema_owners', nargs='+', help='', arg_group='Management')
+        c.argument('manifest_owners', nargs='+', help='', arg_group='Management')
+        c.argument('incident_routing_service', type=str, help='', arg_group='Management')
+        c.argument('incident_routing_team', type=str, help='', arg_group='Management')
+        c.argument('incident_contact_email', type=str, help='', arg_group='Management')
```
Comment (fengzhou-msft) [inline src/providerhub/azext_providerhub/generated/_params.py:194]: I see a lot of empty help messages for the arguments. Please add the right messages for them.

File: src/providerhub/azext_providerhub/generated/_params.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        c.argument('linked_access_checks', action=AddLinkedAccessChecks, nargs='+', help='')
+        c.argument('default_api_version', type=str, help='')
+        c.argument('logging_rules', type=validate_file_or_dict, help=' Expected value: json-string/@json-file.')
+        c.argument('throttling_rules', action=AddThrottlingRules, nargs='+', help='')
+        c.argument('required_features', nargs='+', help='')
+        c.argument('enable_async_operation', arg_type=get_three_state_flag(), help='')
+        c.argument('provisioning_state', arg_type=get_enum_type(['NotSpecified', 'Accepted', 'Running', 'Creating',
+                                                                 'Created', 'Deleting', 'Deleted', 'Canceled',
+                                                                 'Failed', 'Succeeded', 'MovingResources',
+                                                                 'TransientFailure', 'RolloutInProgress']), help='')
```
Comment (fengzhou-msft) [inline src/providerhub/azext_providerhub/generated/_params.py:270]: `provisioning_state` should be a readonly property and should be removed in create command. Please also check your swagger file and you may need to fix it as well.

File: src/providerhub/azext_providerhub/generated/_params.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
```
Comment (fengzhou-msft) [inline src/providerhub/azext_providerhub/generated/_params.py:318]: remove `provisioning_state` in update command.

File: src/providerhub/azext_providerhub/generated/_params.py [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+                   'begins in rest of the world two.')
+        c.argument('skip_regions', action=AddSkipRegions, nargs='+', help='The canary regions to skip.')
+
+    with self.argument_context('providerhub default-rollout update') as c:
+        c.argument('provider_namespace', type=str,
+                   help='The name of the resource provider hosted within ProviderHub.')
+        c.argument('rollout_name', type=str, help='The rollout name.')
+        c.argument('row_two_wait_duration', type=str, help='The wait duration before the rollout begins '
+                   'in rest of the world two.')
+        c.argument('skip_regions', action=AddSkipRegions, nargs='+', help='The canary regions to skip.')
```
[the remark this answers is not in the document]
Comment (wwendyc) [reply src/providerhub/azext_providerhub/generated/_params.py]: It is possible for users to input an value for the skip_regions if that's what you meant by clearing? Updated nargs tp *

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**